### DISCUSSION TITLE
Configuration option stabilization

### DISCUSSION THREAD

**Discussion by calebcartwright** [rust-lang/rustfmt#5365] [Q&A] [answered]
This is intended to cover the variants of typical questions relative to the stabilization of config options, such as:

How are config options stabilized?
When will a config option be stabilized?
Why is a config option not stabilized?

etc.

**Comment by calebcartwright** [comment, marked as answer]
We have a [process we follow for stabilizing configuration options](https://github.com/rust-lang/rustfmt/blob/master/Processes.md#conditions) which enumerates the thresholds an option has to be able to clear in order to be ready for stabilization. All unstable options are presumed to need to remain unstable unless and until the option has been explicitly demonstrated to clear those thresholds.

This holds true regardless of whether a config option is 5 days old or 5 years old.

**Reply by daniel-wong-dfinity-org** [reply]
> Shipping is a feature.
> --Eric Schmidt, former CEO, Google
